packages/ui/src/components/Command/AiCommand.tsx

```diff
@@ -131,9 +131,96 @@ export function useAiChat({
   const [isResponding, setIsResponding] = useState(false)
   const [hasError, setHasError] = useState(false)
 
-  const [currentMessageIndex, setCurrentMessageIndex] = useState(1)
   const [messages, dispatchMessage] = useReducer(messageReducer, [])
 
+  const router = useRouter()
+  const projectRef = router.query.ref
+
+  const [isInputDisabled, setIsInputDisabled] = useState(false)
+  const [submittedFeedback, setSubmittedFeedback] = useState<'yes' | 'no' | null>(null)
+
+  const handleFeedback = (feedback: 'yes' | 'no') => {
+    dispatchMessage({
+      type: 'update',
+      message: {
+        supportFirstPrompt: false,
+      },
+    })
+
+    if (feedback === 'no') {
+      dispatchMessage({
+        type: 'update',
+        message: {
+          supportSecondPrompt: 'no',
+        },
+      })
+    } else {
+      setIsInputDisabled(true)
+      dispatchMessage({
+        type: 'update',
+        message: {
+          supportSecondPrompt: undefined,
+        },
+      })
+      dispatchMessage({
+        type: 'new',
+        message: {
+          status: MessageStatus.Complete,
+          role: MessageRole.Assistant,
+          content: 'Great to hear I could help you resolve your issue!',
+        },
+      })
+    }
+  }
+
+  const handleSubmittedFeedback = (
+    submittedFeedback: 'Ask another question' | 'Please get me in touch with Supabase Support'
+  ) => {
+    dispatchMessage({
+      type: 'update',
+      message: {
+        supportFirstPrompt: false,
+      },
+    })
+
+    dispatchMessage({
+      type: 'update',
+      message: {
+        supportSecondPrompt: undefined,
+      },
+    })
+
+    if (submittedFeedback === 'Please get me in touch with Supabase Support') {
+      setIsInputDisabled(true)
+      dispatchMessage({
+        type: 'new',
+        message: {
+          status: MessageStatus.Complete,
+          role: MessageRole.Assistant,
+          content: 'Redirecting you to the support form. Please wait...',
+        },
+      })
+
+      const userFirstReply = messages.find(({ role }) => role === MessageRole.User)?.content || ''
+
+      setTimeout(() => {
```

Comment:
I've seen other support AI agents often do the same thing, and I wanted to avoid any confusion by making the redirection too sudden for the user. Makes sense?